File: app/src/main/java/ai/anaha/signup/hlsplayer/fragments/PlayBackSpeedViewFragment.java
package ai.anaha.signup.hlsplayer.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import ai.anaha.signup.hlsplayer.AnahaPlayerActivity;
import ai.anaha.signup.hlsplayer.R;
import ai.anaha.signup.hlsplayer.adapters.PlaybackSpeedAdapter;

public final class PlayBackSpeedViewFragment extends Fragment {

    private final AnahaPlayerActivity.PlaybackSpeedListener playbackSpeedListener;
    private float playbackSpeed;

    public PlayBackSpeedViewFragment(float playbackSpeed, AnahaPlayerActivity.PlaybackSpeedListener playbackSpeedListener) {
        // Retain instance across activity re-creation to prevent losing access to init data.
        this.playbackSpeed = playbackSpeed;
        this.playbackSpeedListener = playbackSpeedListener;
        setRetainInstance(true);
    }

    @Override
    public View onCreateView(
            LayoutInflater inflater,
            @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        View rootView =
                inflater.inflate(
                        R.layout.custom_exo_track_selection_dialog, container, /* attachToRoot= */ false);
        rootView.findViewById(R.id.tvTitle).setVisibility(View.GONE);
        PlaybackSpeedAdapter playbackSpeedAdapter = new PlaybackSpeedAdapter(
                container.getContext().getResources().getStringArray(R.array.exo_playback_speeds),
                container.getContext().getResources().getIntArray(R.array.exo_speed_multiplied_by_100),
                playbackSpeedListener);
        playbackSpeedAdapter.updateSelectedIndex(playbackSpeed);
        RecyclerView settingsView = rootView.findViewById(R.id.exo_settings_listview);
        settingsView.setAdapter(playbackSpeedAdapter);
        settingsView.setLayoutManager(new LinearLayoutManager(getContext()));
        settingsView.suppressLayout(false);
        return rootView;
    }
}

File: HLSPlayer/src/main/java/ai/anaha/hlsplayer/fragments/PlayBackSpeedViewFragment.kt
package ai.anaha.hlsplayer.fragments

import ai.anaha.hlsplayer.HLSPlayerActivity.PlaybackSpeedListener
import ai.anaha.hlsplayer.R
import ai.anaha.hlsplayer.adapters.PlaybackSpeedAdapter
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class PlayBackSpeedViewFragment(
    private val playbackSpeed: Float,
    private val playbackSpeedListener: PlaybackSpeedListener
) : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val rootView = inflater.inflate(
            R.layout.custom_exo_track_selection_dialog, container,  /* attachToRoot= */false
        )
        rootView.findViewById<View>(R.id.tvTitle).visibility = View.GONE
        val playbackSpeedAdapter = PlaybackSpeedAdapter(
            container!!.context.resources.getStringArray(R.array.exo_playback_speeds),
            container.context.resources.getIntArray(R.array.exo_speed_multiplied_by_100),
            playbackSpeedListener
        )
        playbackSpeedAdapter.updateSelectedIndex(playbackSpeed)
        val settingsView: RecyclerView = rootView.findViewById(R.id.exo_settings_listview)
        settingsView.adapter = playbackSpeedAdapter
        settingsView.layoutManager = LinearLayoutManager(context)
        settingsView.suppressLayout(false)
        return rootView
    }
}